docs/dev/test-architecture.adoc

```diff
@@ -184,7 +184,13 @@ Context("when running help for storage command", func() {
 ----
 +
 
-* Due to parallel test run support make sure that the should run in isolation, otherwise the test result will lead to race condition. To achieve this ginkgo provides some in build functions `BeforeEach`, `AfterEach` etc.
+* Don't create a new test spec for the steps which can be run with the existing specs. Also try to avoid adding the functionality which are part of another feature test file, only add if it is a key requirement.
++ 
+
+* Spec level conditions, pre and post requirements should be run in ginkgo built-in tear down steps `JustBeforeEach` and `JustAfterEach`
```

Comment:
This is confusing.. maybe link to Ginkgo documentation? Even I don't understand this much :(

Reply:
Yes, adding ginkgo documentation seems hard and fast requirement but IMO we should add ginkgo link to the very first line of test guidelines. TBH adding the link for every steps doesn't make sense to me.

Reply:
@prietyc123 No need to give the full doc link, you should provide the link to the teardown feature just make the test guideline intutive. For example

Spec level conditions, pre and post requirements should be run in ginkgo built-in tear down  link:http://onsi.github.io/ginkgo/#structuring-your-specs[steps] `JustBeforeEach` and `JustAfterEach`

Reply:
@amitkrout But its already there https://github.com/openshift/odo/pull/3726/files#diff-27813ce26d5cb87e1c303086adb6db24R141 before jotting down the key points. Does it make sense of referring same doc everytime. AFAIU users would anyway get the idea from the referred doc before going through further points.

Reply:
@prietyc123 https://onsi.github.io/ginkgo/#getting-ginkgo is the whole ginkgo doc and http://onsi.github.io/ginkgo/#structuring-your-specs[steps] is a quick reference of that particular steps. These teardown steps are used more frequently in the test suite, so for a contribute it will be like a first look reference instead of going through the whole doc.

Reply:
Check for your reference - https://github.com/openshift/odo/pull/3726/files#diff-27813ce26d5cb87e1c303086adb6db24R270.